src/solvers/mp/RCPricer.cpp

```diff
@@ -168,160 +151,77 @@ vector<MyVar*> RotationPricer::pricing(double bound, bool before_fathom){
 	SubProblem* subProblem;
-	map<const Contract*, SubProblem*>::iterator it =  subProblems_.find(pNurse->pContract_);
+	auto it = subProblems_.find(pNurse->pContract_);
 	// Each contract has one subproblem. If it has not already been created, create it.
 	if( it == subProblems_.end() ){
-		subProblem = new SubProblem(pScenario_, nbDays_, pNurse->pContract_, pMaster_->pInitState_);
-		subProblems_.insert(it, pair<const Contract*, SubProblem*>(pNurse->pContract_, subProblem));
+	  if (shortSubproblem_)
+	    subProblem = new SubProblemShort(pScenario_, nbDays_, pNurse->pContract_, pMaster_->pInitialStates());
+	  else
+	    subProblem = new SubProblem(pScenario_, nbDays_, pNurse->pContract_, pMaster_->pInitialStates());
+	  // then build the rcspp
+	  subProblem->build();
+	    
+	  //	  subProblem = new SubProblem(pScenario_, nbDays_, pNurse->pContract_, pMaster_->pInitState_, noShort);
+		subProblems_[pNurse->pContract_] = subProblem;
 	} else {
 		subProblem = it->second;
 	}
 	return subProblem;
 }
 
 // Add the rotations to the master problem
-void RotationPricer::addRotationsToMaster(){
-
-    // COMPUTE THE COST OF THE ROTATIONS
-	for(Rotation& rot: newRotationsForNurse_){
-		rot.computeCost(pScenario_, pMaster_->pPreferences_, pMaster_->theLiveNurses_,nbDays_);
-		rot.treeLevel_ = pModel_->getCurrentTreeLevel();
-	}
-
-	// SORT THE ROTATIONS
-	sortNewlyGeneratedRotations();
+int RCPricer::addColumnsToMaster(int nurseId){
+	// SORT THE SOLUTIONS
+  sortNewlyGeneratedSolutions();
 
 	// SECOND, ADD THE ROTATIONS TO THE MASTER PROBLEM (in the previously computed order)
-	int nbRotationsAdded = 0;
-	for(Rotation& rot: newRotationsForNurse_){
-		allNewColumns_.push_back(pMaster_->addRotation(rot, baseName));
-		++nbRotationsAdded;
-		// DBG
-		//cout << rot.toString(nbDays_) << endl;
-		if(nbRotationsAdded >= nbMaxRotationsToAdd_)
+	int nbcolumnsAdded = 0;
+	for(const RCSolution& sol: newSolutionsForNurse_){
+		allNewColumns_.emplace_back(pMaster_->addColumn(nurseId, sol));
+		++nbcolumnsAdded;
+		if(nbcolumnsAdded >= nbMaxColumnsToAdd_)
 			break;
 	}
+
+  return nbcolumnsAdded;
 }
 
 // Sort the rotations that just were generated for a nurse. Default option is sort by increasing reduced cost but we
 // could try something else (involving disjoint columns for ex.)
-void RotationPricer::sortNewlyGeneratedRotations(){
-	std::stable_sort(newRotationsForNurse_.begin(), newRotationsForNurse_.end(), Rotation::compareDualCost);
+void RCPricer::sortNewlyGeneratedSolutions(){
+	std::stable_sort(newSolutionsForNurse_.begin(), newSolutionsForNurse_.end(),
+	    [](const RCSolution& sol1, const RCSolution& sol2) { return sol1.cost < sol2.cost; });
 
 }
 
 // Set the subproblem options depending on the parameters
 //
-//void RotationPricer::setSubproblemOptions(vector<SolveOption>& options, int& maxRotationLengthForSubproblem,
-//		LiveNurse* pNurse){
+//void RCPricer::setSubproblemOptions(vector<SolveOption>& options, int& maxRotationLengthForSubproblem,
```

Comment:
Why is this commented? Delete or store in an issue?

Reply:
deleted as the pricer does not store it but just passed to the subproblem.